eplus_rmd/status_reporter.py

```diff
@@ -0,0 +1,81 @@
+from yaml import safe_load
+from pathlib import Path
+from datetime import datetime
+
+
+class StatusReporter:
+    def __init__(self):
+        self.extra_schema = {}
+        parent_dir = Path(__file__).parent
+
+        # the extra schema file includes extra tags on fields related to appendix G and energyplus
+        extra_schema_file = 'ASHRAE229_extra.schema.yaml'
+        extra_schema_path = Path(parent_dir).joinpath(extra_schema_file)
+        if extra_schema_path.exists():
+            with open(extra_schema_path) as schema_f:
+                self.extra_schema = safe_load(schema_f)
+        report_file = 'energyplus_implementation_report.txt'
+        self.report_file_path = Path(parent_dir).joinpath(report_file)
+
+    def generate(self, rmd_dict):
+        if self.extra_schema:  # if the YAML schema file is not present then don't generate report
+            with open(self.report_file_path, 'w') as f:
+                f.write('============= Generated Report ==============\n')
```

Comment:
To avoid all the hardcoded newline characters, you can use python's print function and it will by default have a newline terminator:

```
print('something here', file=f)
```